lib/downloadresponse.php

```diff
@@ -30,7 +30,10 @@ public function __construct(IRequest $request, $user, $path) {
 		$this->user = $user;
 		$this->path = $path;
 		
-		$this->view = new View('/' . $user);
+		$data_dir = \OC::$server->getConfig()->getSystemValue('datadirectory', \OC::$SERVERROOT . '/data');
+		$fake_root = str_replace($data_dir, '', \OC::$server->getUserManager()->get($user)->getHome());
```

Comment:
This approach is wrong and wont work if the home folder is outside the datadir or when the home folder isn't local at all.

Paths within ownCloud's virtual filesystem **always** follow `/$userid/files/....` no matter what the actual physical location of the home folder is.

The correct way to fix this is to make sure the filesystem is setup for the user.